conda/cli/main_config.py

```diff
@@ -355,22 +375,45 @@ def print_config_item(key, value):
                 stdout_write(" ".join(("--add", key, repr(item))))
 
 
+def _key_exists(key: str, warnings: list[str], context=None) -> bool:
+    """
+    Logic to determine if the key is a valid setting.
+    """
+    if context is None:
+        from ..base.context import context
+
+    first, *rest = key.split(".")
+    if (
+        first == "plugins"
+        and len(rest) > 0
+        and rest[0] in context.plugins.parameter_names
+    ):
+        return True
+
+    if first not in context.list_parameters():
```

Comment:
Why do we use `Configuration.parameter_names` for plugins but `Configuration.list_parameters()` for non-plugins?

Reply:
We could switch it to use `list_parameters` instead. I'll do it.

Reply:
This must have been an oversight because `list_parameters` was being used everywhere else in that module.

Reply:
Address this in my latest commit.

Reply:
Just a thought, does this mean that parameters cannot be looked up by their aliases?

Reply:
Yeah, that's true, but you currently cannot do this with normal parameters either, so I'm not worried about it. This is something that should be addressed with a separate pull request.